extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/api/RegistrationService.java

```diff
@@ -46,6 +66,36 @@ public List<Participant> listParticipants() {
 
     public void addParticipant(Participant participant) {
         monitor.info("Adding a participant in the dataspace.");
-        participantStore.addParticipant(participant);
+        participantStore.save(participant);
+    }
+
+    public void start() {
+        stateMachine.start();
+    }
+
+    public void stop() {
+        stateMachine.stop();
+    }
+
+    private Boolean processOnboardingInitiated(Participant participant) {
+        participant.transitionAuthorizing();
+        participantStore.save(participant);
+        return true;
+    }
+
+    private Boolean processAuthorizing(Participant participant) {
+        var verificationResult = credentialsVerifier.verifyCredentials();
```

Comment:
nit: I'd change this var name to eg. `verified` or `verifiedCredentials` then we read the code as a sentence: if verfied then...